samples/xamarin-forms/LoginApp/LoginApp/ViewModels/SignUpViewModel.cs

```diff
@@ -0,0 +1,99 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using System.Reactive;
+using System.Reactive.Disposables;
+using Acr.UserDialogs;
+using LoginApp.ViewModels.Abstractions;
+using ReactiveUI;
+using ReactiveUI.Fody.Helpers;
+using ReactiveUI.Validation.Abstractions;
+using ReactiveUI.Validation.Contexts;
+using ReactiveUI.Validation.Extensions;
+using Splat;
+
+namespace LoginApp.ViewModels
+{
+    public class SignUpViewModel : ViewModelBase, ISupportsValidation
+    {
+        private readonly Func<string, bool> _isDefined = value => !string.IsNullOrEmpty(value);
+        private readonly IUserDialogs _dialogs;
+
+        [Reactive] public string UserName { get; set; }
+
+        [Reactive] public string Password { get; set; }
+
+        [Reactive] public string ConfirmPassword { get; set; }
+
+        public ReactiveCommand<Unit, bool> SignUp { get; }
+
+        public ValidationContext ValidationContext { get; } = new ValidationContext();
+
+        public SignUpViewModel(IScreen hostScreen = null, IUserDialogs dialogs = null)
+            : base(hostScreen)
+        {
+            _dialogs = dialogs ?? Locator.Current.GetService<IUserDialogs>();
+            SignUp = ReactiveCommand.Create(SignUpImpl, this.IsValid());
+            this.WhenActivated(disposables =>
```

Comment:
Probably worth removing this WhenActivated block. The view model isn't referencing anything other than itself, so there is no potential for a memory leak -> WhenActivated and DisposeWith aren't necessary.